### ISSUE TITLE
Remailer software missing

### ISSUE THREAD

**Issue by denishi** [nylira/prism-break#1469]
I'm missing anonymous remailers, particularly the **Mixmaster** network, as the most secure method of anonymous mail transmission.

With the **OmniMix** Windows SMTP/POP3/NNTP proxy server usual client software can take part in anonymous remailing. For maximum transparency there's also an IDE setup program available including all sources to build the executable automatically from scratch.

http://en.wikipedia.org/wiki/Anonymous_remailer
http://www.danner-net.de/om.htm

**Comment by Zegnat** [comment]
* Mixmaster seems to have a proprietary licence though, and will not be accepted on PRISM Break.
* I can’t find OmniMix’ sources or licence anywhere. (They might be part of that IDE download, but I am not running an untrusted executable to find out.)

Interesting idea for a category, but we would need to find some software to put in it still.

**Comment by denishi** [comment]
> Mixmaster seems to have a proprietary licence, and will not be accepted on PRISM Break.

What's wrong with https://github.com/merkinmuffley/mixmaster4096/blob/master/COPYRIGHT ?

> I can’t find OmniMix’ source or licence anywhere. (They might be part of that IDE download, but I am not running an untrusted executable to find out.)

To view the contents you don't have to run it, http://www.danner-net.de/omom/tutorinstall.htm says:

*You may be interested in getting an impression of the files to be installed without having to run the setup program. In that case extract its contents for example using the [Inno Setup Unpacker](http://sourceforge.net/projects/innounp/):*

*c:\InnoUnpack>innounp.exe -x -d"OmniMix Setup Files" OmniMix_2.1.2_Uno_Setup.exe*

Of course innounp also unpacks http://www.danner-net.de/om/OmniMix_IDE_2.1.3_Uno_Setup.exe.  But be careful.  I can't find the innounp source code above v0.35 on their website!

> Interesting idea for a category, but we would need to find some software to put in it still.

I'm sorry, but that's all we have.  OTOH Mixmaster and OmniMix are completely transparent by granting you full source code access to build the binaries yourself.  Do you know of any other Windows application, which comes along with a click-to-build IDE even grandma can run?  I'd like to have such for a lot of FOSS Windows security programs.

Concerning security, I see no tool on your list being more resilient against SIGINT operations done by global adversaries like the Five Eyes Alliance.  A combination of Tor (unfortunately circuits built of no more than 3 nodes and prone to correlation attacks) plus remailers (providing latency), which OmniMix deploys, is the best we have when aiming at undetectable truly anonymous communication.  Nevertheless it's a legit strategy to focus strictly on software free of any copyright and sometimes accept weak spots.

But, while celebrating such uncompromising purism, why on earth do you list external mail services, where you're completely and utterly at the operators' mercy?  Don't get me wrong, I'm not questioning their integrity.  They are certainly doing a great job.  However, haven't we recently learned the hard way, that nobody ought to be trusted?

So even if you still prioritize the FOSS criterion over security features, you may list not only the bad apples, but also outstanding privacy software like Mixmaster, OmniMix and Moxie's Signal as proprietary alternatives in their respective categories (here "Anonymizing Networks" / "Instant Messaging").  Only this way users see the big picture to make appropriate decisions.  You're obviously in need of another evaluation level between potentially compromized and FOSS.  Thanks. :wink:

**Comment by Zegnat** [comment]
> > Mixmaster seems to have a proprietary licence, and will not be accepted on PRISM Break.
> 
> What's wrong with https://github.com/merkinmuffley/mixmaster4096/blob/master/COPYRIGHT ?

I don’t recognise that one as any FSF or OSI approved licence. In fact, I did find some discussion from Debian mailing lists about how their licence did not comply with Debian rules for free software. I am not sure if this was ever addressed or not.

PRISM Break has made the conscious decision to only list projects with FSF (or, as border-line accepted, OSI) compliant licenses. Mostly because we do not ourselves want to spit through a licence to see if it is truly libre.

> Of course innounp also unpacks http://www.danner-net.de/om/OmniMix_IDE_2.1.3_Uno_Setup.exe. But be careful. I can't find the innounp source code above v0.35 on their website!

Good tip. I am not sure where you were looking, [innoextract](https://constexpr.org/innoextract/) has a tarball of 1.5 right there on the homepage. It is also available through package managers (in my case Homebrew).

So here is [the OmniMix licence file](https://gist.github.com/Zegnat/906490601cf4ff78d0a8) for everyone to look at. I can right now say it will not be included on PRISM Break. It is not open-source in any way. You are not allowed to create derivative works at all.

> Do you know of any other Windows application, which comes along with a click-to-build IDE even grandma can run? I'd like to have such for a lot of FOSS Windows security programs.

Why? My grandmother would literally gain no advantage from such a tool. Only people who can read and understand the source code are helped. And most of these people already have compilers installed and are perfectly happy with a tarball. A custom IDE does not give me anything. I don’t even know if it will use a trusted compiler, unless I also have access to the source of the IDE!

The reason you want to offer source to everyone for local compiling is so they know nothing has been added on top of it. If you have a trusted compiler installed (this is always debatable) and are downloading raw source from a trusted source, you might not have to be able to read the code yourself (this would help grandma). Any other ways of packaging source code is just fluff.

> You're obviously in need of another evaluation level between potentially compromized and FOSS.

Maybe. But FOSS means anyone is allowed to spread it around. How will you spread these proprietary programs to Asia? You are not allowed to sell CDs, so you might have to incur huge losses on logistics. You are also not allowed to translate them, because no modifications are allowed.

I would like to flip this around: why don’t Mixmaster and OmniMix change to a truly open source licence?

> […] Moxie's Signal as proprietary alternatives

Open Whisper Systems’ Signal is open source, I don’t know why you think otherwise. We do not list it because it has chosen to depend on Google for their notifications. PRISM Break doesn’t want to introduce Google dependencies with the projects that are recommended. That is all.

---

* According to the current guidelines for PRISM Break, neither Mixmaster nor OmniMix qualify. That is pretty clear-cut. @nylira (and the community) will need to decide to change the guidelines for them to be accepted.
* I still like the idea of getting remailer software listed on PRISM Break. It is a good idea for a category. And I would love to see if anyone knows any other software.

Cheers!
